src/metabase/driver/mysql.clj

```diff
@@ -48,11 +48,7 @@
 
 (defmethod driver/display-name :mysql [_] "MySQL")
 
-(defmethod driver/database-supports? [:mysql :nested-field-columns] [_ _ database]
-  (let [json-setting (get-in database [:details :json-unfolding])]
-    (if (nil? json-setting)
-      true
-      json-setting)))
+(defmethod driver/database-supports? [:mysql :nested-field-columns] [_driver _feat _db] true)
```

Comment:
This is part of a refactor. Now, `database-supports?` just returns a boolean based on the driver. The `database.details.json-unfolding` setting is checked by the caller of this function separately if needed, e.g. here https://github.com/metabase/metabase/pull/28742/files#diff-e4db3baaafcba181fc735b94c34048deb437fc47e8a2bd3acb9057732d92b330R159-R161

Reply:
are we sure this will be true by default? json only supports from MySQL 5.7.8, I think we want to support 5.7?

Reply:
If that's the case, why do the JSON tests pass for MySQL? We're using 5.7.23 in CI

Reply:
I checked our docs, and you're right. We're meant to support MySQL 5.7.7, which doesn't have the JSON type. It's a pity we're not testing against this in CI. But since JSON unfolding will only apply to JSON types, it won't cause any bugs. It's the same scenario as redshift was in until today, with [this PR](https://github.com/metabase/metabase/pull/29465). The only bug visible to users (which currently exists) would be that the "Unfold JSON by Default" setting would be present in the MySQL DB's connection settings page, regardless of its version.

Reply:
I added a comment explaining this https://github.com/metabase/metabase/commit/e83d67498f699b504b0a8e93ea46fd65033fc24e

Reply:
FWIW

5.7 will EOL in 10 months.

https://endoflife.date/mysql